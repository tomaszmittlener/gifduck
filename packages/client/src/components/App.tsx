import React, { useEffect, useState } from 'react'
import Text from './Text'

const API_URL = 'http://localhost:4000/api/v1/test'

export default function App() {
  const [data, setData] = useState<string>('')

  const getData = async () => {
    try {
      const response = await fetch(API_URL)
      const { data } = await response.json()
      setData(data)
    } catch (e) {
      throw new Error(e)
    }
  }

  // let number = 1

  const number = 'test'

  useEffect(() => {
    getData()

    console.log(number)
    // console.log(Test())
  }, [])

  return (
    <div>
      <Text value={'React app with Typescript'} />
      <br />
      <br />
      <Text value={data} />
    </div>
  )
}
