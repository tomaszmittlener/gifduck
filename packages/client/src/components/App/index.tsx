import React, { FunctionComponent } from 'react'
import { ThemeProvider } from 'styled-components'
import { BrowserRouter as Router } from 'react-router-dom'

import GlobalStyles from 'styles/GlobalStyles'
import Theme from 'styles/theme'
import Search from 'views/Search'

const App: FunctionComponent = () => {
  return (
    <Router>
      <ThemeProvider theme={Theme}>
        <GlobalStyles />
        <Search />
      </ThemeProvider>
    </Router>
  )
}

export default App
