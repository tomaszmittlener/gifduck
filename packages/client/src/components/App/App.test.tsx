import React from 'react'
import { render, cleanup } from '@testing-library/react'

import App from './'

afterEach(cleanup)

describe('App', () => {
  test('renders correctly', async () => {
    const { container } = render(<App />)

    expect(container).toMatchSnapshot()
  })
})
