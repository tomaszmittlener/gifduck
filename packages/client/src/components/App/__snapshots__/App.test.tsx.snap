// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders correctly 1`] = `
.c0 {
  color: red;
  font-weight: bold;
}

.c1 {
  color: blue;
  font-weight: bold;
}

<div>
  <div>
    <span
      class="c0"
      color="red"
    >
      React app with Typescript
    </span>
    <br />
    <br />
    <span
      class="c1"
      color="blue"
    />
  </div>
</div>
`;
